applications/mne_rt_server/plugins/neuromag/shmemsocket.cpp

```diff
@@ -370,9 +370,11 @@ int ShmemSocket::disconnect_client ()
 {
     int sock = m_iShmemSock;
     int id = m_iShmemId;
-
     int result = connect_disconnect(sock,-id);
-    this->close_socket ();
+    if (result != FAIL)
```

Comment:
Maybe output a warning if the connect_disconnect returns FAIL. Use QWarning for this. `qWarning() << "";`